manifests/c/Canon/i-SENSYS/MF4010-MF4018/2.10/Canon.i-SENSYS.MF4010-MF4018.installer.yaml

```diff
@@ -0,0 +1,18 @@
+# Created using wingetcreate 1.5.7.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: Canon.i-SENSYS.MF4010-MF4018
+PackageVersion: "2.10"
+InstallerType: exe
+InstallerSwitches:
+  Silent: /qn
+  SilentWithProgress: /qn
+Installers:
+- Architecture: x64
+  InstallerUrl: https://gdlp01.c-wss.com/gds/5/0100004725/03/MF4010_MFDrivers_W64_uk_EN.exe
+  InstallerSha256: C5E0C5800DE336A886D3A00666C6FE63349F806C56D4C5760C7CCFC2914A89AA
+  NestedInstallerType: exe
+  NestedInstallerFiles: Setup.exe
+    - RelativeFilePath: x64

```

Comment:
```suggestion
InstallerType: zip
InstallerSwitches:
  Silent: /qn
  SilentWithProgress: /qn
Installers:
- Architecture: x64
  InstallerUrl: https://gdlp01.c-wss.com/gds/5/0100004725/03/MF4010_MFDrivers_W64_uk_EN.exe
  InstallerSha256: C5E0C5800DE336A886D3A00666C6FE63349F806C56D4C5760C7CCFC2914A89AA
  NestedInstallerType: exe
  NestedInstallerFiles: 
    - RelativeFilePath: x64\Setup.exe
```

Reply:
I tried this a few times to figure out the right pattern. First, my earlier suggestion wasn't actually valid YAML. I believe `NestedInstallerFiles` shouldn't have any text following it, to represent an array field. So moving the installer name into the `RelativeFilePath` parameter allows the package manager to parse this section. 

On most runs, the installer switches were applied to the outer EXE. This caused only empty directories to become extracted, intsead of the driver files. A feature here is that installer switches are supposed to run against the nested installer, not the outer installer. Based on this, I'm not sure if we support EXE within EXE. 

The installer is a self-extracting ZIP, so I guessed that treating it as one might allow the inner installer to be handled correctly. And this seems to be the case. Manual Validation is still underway, but the driver files appear to have extracted correctly, and shows up in Apps Settings: 
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/b808541f-7d4d-4d55-aaa8-8c934c234474)